src/main/java/com/rtb/projectmanagementtool/project/ProjectData.java

```diff
@@ -41,32 +52,57 @@ public ProjectData(String name, String description, long creatorId) {
    */
   public ProjectData(Entity entity) {
     this.id = (Long) entity.getKey().getId();
+    this.creatorId = (Long) entity.getProperty(PROPERTY_CREATOR);
     this.name = (String) entity.getProperty(PROPERTY_NAME);
     this.description = (String) entity.getProperty(PROPERTY_DESCRIPTION);
 
-    // Tasks
-    Object entityProperty;
-    if ((entityProperty = entity.getProperty(PROPERTY_TASKS)) != null) {
-      this.tasks = (ArrayList<Long>) entityProperty;
-    } else {
-      this.tasks = new ArrayList<Long>();
+    // Instantiate containers
+    this.tasks = new HashSet<Long>();
+    this.users = new HashMap<UserProjectRole, HashSet<Long>>();
+    this.users.put(UserProjectRole.ADMIN, new HashSet<Long>());
+    this.users.put(UserProjectRole.MEMBER, new HashSet<Long>());
+
+    // Update containers with entity properties
+    parseEntityHashSets(entity, PROPERTY_TASKS);
+    parseEntityHashSets(entity, PROPERTY_ADMINS);
+    parseEntityHashSets(entity, PROPERTY_MEMBERS);
+  }
+
+  /**
+   * Used by ProjectData(Entity) to parse entity properties into respective HashSets of this class
+   *
+   * @param entity the entity
+   * @param propertyName the name of entity property
+   */
+  private void parseEntityHashSets(Entity entity, String propertyName) {
+    Object entityProperty = entity.getProperty(propertyName);
+    if (entityProperty == null) {
+      return;
     }
 
-    // Users
-    if ((entityProperty = entity.getProperty(PROPERTY_USERS)) != null) {
-      this.users = (ArrayList<String>) entityProperty;
-    } else {
-      this.users = new ArrayList<String>();
+    Collection collection = (Collection) entityProperty;
+    Iterator<Long> iterator = collection.iterator();
+
+    while (iterator.hasNext()) {
+      if (propertyName.equals(PROPERTY_ADMINS)) {
+        addUser(UserProjectRole.ADMIN, iterator.next());
+      } else if (propertyName.equals(PROPERTY_MEMBERS)) {
+        addUser(UserProjectRole.MEMBER, iterator.next());
+      } else if (propertyName.equals(PROPERTY_TASKS)) {
+        addTask(iterator.next());
+      }
     }
   }
 
   /** @return the entity representation of this class */
   public Entity toEntity() {
     Entity entity = new Entity("Project");
     entity.setProperty(PROPERTY_NAME, this.name);
+    entity.setProperty(PROPERTY_CREATOR, this.creatorId);
     entity.setProperty(PROPERTY_DESCRIPTION, this.description);
     entity.setProperty(PROPERTY_TASKS, this.tasks);
-    entity.setProperty(PROPERTY_USERS, this.users);
+    entity.setProperty(PROPERTY_ADMINS, this.users.get(UserProjectRole.ADMIN));
+    entity.setProperty(PROPERTY_MEMBERS, this.users.get(UserProjectRole.MEMBER));
```

Comment:
If admins and members are stored as two separate properties in an entity in datastore, is there a reason for combining them in the ProjectData class as a hashmap instead of having an admins set and a members set?

Reply:
Maintain use of enum